src/main/java/org/jenkinsci/test/acceptance/guice/World.java

```diff
@@ -62,14 +64,16 @@ private void startTestScope(Injector i, String testName) {
         i.getInstance(TestName.class).testName = testName;
     }
 
-    public void endTestScope() {
-        injector.getInstance(TestCleaner.class).performCleanUp();
+    public void endTestScope() throws Exception {
+        List<Throwable> errors = injector.getInstance(TestCleaner.class).performCleanUp();
         injector.getInstance(TestLifecycle.class).endTestScope();
 
         for (SubWorld sw : subworlds.list(injector)) {
-            sw.injector.getInstance(TestCleaner.class).performCleanUp();
+            errors.addAll(sw.injector.getInstance(TestCleaner.class).performCleanUp());
             sw.injector.getInstance(TestLifecycle.class).endTestScope();
         }
+
+        MultipleFailureException.assertEmpty(errors);
```

Comment:
This is one of three callers of `performCleanup`. Here we propagate the errors. The other two callers are the world cleaner thread and connection handling thread, both of which have no way of propagating the errors and as such ignore the return value of `performCleanup()`, unlike what we are doing here.